### PR TITLE
Containerize Libvirt

### PR DESCRIPTION
**Pull request by stu-gott**
Provide Libvirtd inside a container. The intention is that
/var/run/libvirt will be bind mounted from the host, and optionally
Ports 16509/5900+ if TCP connections are required for libvirt or
spice/vnc.

### CODE REVIEW

**Comment by stu-gott** [comment]
I'm conflicted on one point. I put the Dockerfile and components at the toplevel of this git repo. Is that the correct approach, or should we put them into a subfolder?

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,7 @@
+vendor/*
```
Comment (rmohr) [inline .gitignore]: You can clean up this file

File: README.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+Note: Make sure to not run libvirtd on the host.
+
+Start the container
+
+    docker run \
+      --name libvirtd \
+      --rm \
+      --net=host \
+      --pid=host \
+      --user=root \
```
Comment (rmohr) [inline README.md]: do we need --ipc=host too?

File: README.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+These environment variables can be passed into the container
+
+* LIBVIRTD_DEFAULT_NETWORK_DEVICE: Set it to an existing device
+  to let the default network point to it.
+
+# Notes
+
+Considerations that need to be taken into account:
+
+* The D-Bus socket is not exposed inside the container
```
Comment (rmohr) [inline README.md:70]: We could actually provide an example here, on how to expose the dbus socket to the container (something like -v /host/socket:/container/socket)

File: libvirtd.sh [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+mount --rbind /host-sys/fs/cgroup /sys/fs/cgroup
+
+mkdir -p /var/log/kubevirt
+touch /var/log/kubevirt/qemu-kube.log
+chown qemu:qemu /var/log/kubevirt/qemu-kube.log
+
+# We create the network on a file basis to not
+# have to wait for libvirtd to come up
+if [[ -n "$LIBVIRTD_DEFAULT_NETWORK_DEVICE" ]]; then
```
Comment (rmohr) [inline libvirtd.sh:50]: I am not so happy with this section but good enough for now

**Review by rmohr — CHANGES_REQUESTED** [review]
Does this also need virtlogd running somewhere?

In principle we could add a docker compose file which start libvirtd and virtlogd and run tests on travis with qemu full emulation.

**Pull request merged**